src/Shared/Traits.cs

```diff
@@ -104,7 +104,13 @@ internal class EscapeHatches
         /// </summary>
         public readonly bool AlwaysUseContentTimestamp = Environment.GetEnvironmentVariable("MSBUILDALWAYSCHECKCONTENTTIMESTAMP") == "1";
 
-        public readonly bool LogProjectImports = Environment.GetEnvironmentVariable("MSBUILDLOGIMPORTS") == "1";
+        /// <summary>
+        /// Emit events for project imports.
+        /// </summary>
+        /// <remarks>
+        /// This is not cached because the environment variable can be modified inside msbuild.exe
+        /// </remarks>
+        public bool LogProjectImports => Environment.GetEnvironmentVariable("MSBUILDLOGIMPORTS") == "1";
```

Comment:
This is read during every evaluation so we'll need to find solution with the best performance.  My suggestion would to be make this a read/write field and set it in the BinaryLogger where the environment variable is set.

Reply:
Yes, let's avoid reading the environment as much as possible.